src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,236 @@
+                    }else{
+                        const bytes = Array.from(new Uint8Array(reader.result));
+                        base64StringFile = btoa(bytes.map((item) => String.fromCharCode(item)).join(""));
+                    }
+                    if(base64StringFile){
+                        setBase64(base64StringFile)
+                    }
+                }
+            }
+            reader.readAsArrayBuffer(file)
+            setIsDragEnter(false)
+        })
+    }, []);
+    const onDragEnter=useCallback((event:React.DragEvent)=>{
+        event.preventDefault();
+        setIsDragEnter(true);
+    },[]);
+    const onDragLeave=useCallback((event:React.DragEvent)=>{
+        event.preventDefault();
+        setIsDragEnter(false);
+    },[]);
+    const {getRootProps, getInputProps,rootRef} = useDropzone({onDrop,onDragEnter,onDragLeave,disabled})
+
+    return (
+        <Container backgroundColor='white' padding='10px' borderRadius='20px' ref={containerRef} maxHeight={maxHeight} overflow='hidden' height={height} margin={`${margin}px`}>
+            <Container {...getRootProps({dashed:true,withFlexCenter:true,withBorder:!isDragEnter,isDragEnter,padding:'10px',margin:`${margin}px`})}>
+                <SubContainer minHeight={minHeight}>
+                    {isDragEnter?<FileMovingAnimation />:<Icon as={Image} />}
+                    <TextLayout bold color='DarkBlue'>
+                        {title}
+                    </TextLayout>      
+                    <TextLayout size='small' bold color='rgba(128,128,128,.8)'>{subTitle}</TextLayout>
+                    <input {...getInputProps()}  />
+                </SubContainer>
+            </Container>
+            <BottomPanel withBorder padding='30px 20px 43px 20px' opacity={opacityLoading} position={positionLoading} ref={loadingContainerRef} overflow='hidden' width={widthComponent} margin={`${margin}px`} positionTop={positionTopLoading}>
+                <Loading loading={isUploading} message='Uploading...' />
+            </BottomPanel>
+            <BottomPanel opacity={opacityIsSuccess} height={loadingContainerHeight} position={positionIsSuccess} margin={`${margin}px`} positionTop={positionTopLoading} width={widthIsSuccess}>
+                <Container withFlexSpaceBetween>
+                    <TextLayout bold color='DarkBlue'>{successMessage}</TextLayout>
+                    <Icon as={CheckCircle} color={MainTheme.colors.statusColors.green} />
+                </Container>
+            </BottomPanel>
+            <BottomPanel withBorder opacity={opacityIsFailure} height={loadingContainerHeight} position={positionIsFailure} margin={`${margin}px`} positionTop={positionTopLoading} width={widthIsSuccess}>
+                <Container withFlexSpaceBetween>
+                    <TextLayout bold color='DarkBlue'>{failureMessage}</TextLayout>
+                    <Icon as={TimesCircle} color={MainTheme.colors.statusColors.red} />
+                </Container>
```

Comment:
Why aren't you re-using code????

Reply:
I don't know how I can reuse code in this case. BottomPanel renders it's children, which in this case there are three different bottom panel with three different children (there are two similar, but one different, which is the isUploading bottompanel)

Reply:
Children as a prop then?

Reply:
Just inject the things that are different as props

Reply:
Yes, that's it. children it's already a prop. The only thing that occurs to me is to make a react component for the two similar children.